### PR TITLE
Feature/151 project details db

### PR DESCRIPTION
**Pull request by alejandro-castellon**
[AB#151](https://dev.azure.com/funjaladev/7867a1c5-5ac7-4c7f-baff-87d740bf4fc1/_workitems/edit/151)

# Overview

- Inject MongoDB driver with Repository
- Update GetByID with Database

## Outstanding Tasks

-  Unit test
- Swagger Route

### CODE REVIEW

**Comment by pedro-lima-jalafund** [comment]
ShallNoPass

**Review by niver-copa-jalafund — DISMISSED** [review]
I was clone your PR, this working, I don´t find any problems

**Review by niver-copa-jalafund — DISMISSED** [review]

File: src/dal/Jalasoft.TeamUp.Projects.DAL/ProjectsMongoDbRepository.cs [hunk 1]
```diff
@@ -10,26 +10,33 @@
...
         {
-            throw new NotImplementedException();
+            List<Project> projects = collection.Find(x => true).ToList();
+            return projects;
         }
 
         public Project GetById(Guid id)
         {
-            throw new NotImplementedException();
+            Project project = collection.Find(x => x.Id == id).Single();
```
Comment (pedro-lima-jalafund) [inline src/dal/Jalasoft.TeamUp.Projects.DAL/ProjectsMongoDbRepository.cs]: What happens when there isn't a project with that specific GUI?

Comment (alejandro-castellon) [reply src/dal/Jalasoft.TeamUp.Projects.DAL/ProjectsMongoDbRepository.cs]: GetById method returns null so API Controller return a NotFoundObjectResult

**Review by pedro-lima-jalafund — DISMISSED** [review]

File: src/api/Jalasoft.TeamUp.Projects.API/Startup.cs [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-
     public class Startup : FunctionsStartup
     {
         public override void Configure(IFunctionsHostBuilder builder)
         {
             builder.Services.AddScoped<IHealthService, HealthService>();
             builder.Services.AddScoped<IHealthRepository, HealthRepository>();
             builder.Services.AddScoped<IProjectsService, ProjectsService>();
-            builder.Services.AddScoped<IProjectsRepository, ProjectsRepository>();
+            builder.Services.AddScoped<IRepository<Project>, ProjectsMongoDbRepository>();
```
Comment (cristian-chavez-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/Startup.cs:23]: nice update here 👍🏼 that's the intention of the repository pattern

**Review by cristian-chavez-jalafund — DISMISSED** [review]

**Review by pedro-lima-jalafund — APPROVED** [review]

**Review by cristian-chavez-jalafund — APPROVED** [review]

**Review by rodrigo-baldiviezo-jalafund — APPROVED** [review]

**Pull request merged**